breizorro/breizorro.py

```diff
@@ -342,40 +344,100 @@ def load_fits_or_region(filename):
         mask_header['BUNIT'] = 'Jy/beam'
         mask_image = input_image * new_mask_image
         LOGGER.info(f"Number of extended islands found: {len(extended_islands)}")
+        shutil.copyfile(input_file, out_mask_fits)  # to provide a template
+        flush_fits(mask_image, out_mask_fits, mask_header)
+        LOGGER.info("Done")
+        sys.exit(1)
+
+    if args.outcatalog or args.outregion:
+        try:
+            from skimage.measure import find_contours
+        except ImportError:
+            LOGGER.info('pip install breizorro[all] to use cataloguing feature.')
+            exit(1)
+        contours = find_contours(mask_image, 0.5)
+        polygon_regions = []
+        for contour in contours:
+            # Convert the contour points to pixel coordinates
+            contour_pixels = contour
+            # Convert the pixel coordinates to Sky coordinates
+            contour_sky = wcs.pixel_to_world(contour_pixels[:, 1], contour_pixels[:, 0])
+            # Create a Polygon region from the Sky coordinates
+            polygon_region = PolygonSkyRegion(vertices=contour_sky, meta={'label': 'Region'})
+            # Add the polygon region to the list
+            polygon_regions.append(polygon_region)
+        LOGGER.info(f"Number of regions found: {len(polygon_regions)}")
+        if args.outregion:
+            regions.Regions(polygon_regions).write(args.outregion, format='ds9')
+            LOGGER.info(f"Saving regions in {args.outregion}")
+
+    if args.outcatalog and args.imagename:
+        try:
+            import warnings
+            # Suppress FittingWarnings from Astropy
+            # WARNING: The fit may be unsuccessful; check fit_info['message'] for more information. [astropy.modeling.fitting]
+            warnings.resetwarnings()
+            warnings.filterwarnings('ignore', category=UserWarning, append=True)
```

Comment:
I think there's a context manager form (``with something``) to temporarily change warnings settings.